src/main/java/br/com/deveficiente/bolaoapi/services/user/api/UserController.java

```diff
@@ -0,0 +1,68 @@
+package br.com.deveficiente.bolaoapi.services.user.api;
+
+import br.com.deveficiente.bolaoapi.services.user.UserService;
+import br.com.deveficiente.bolaoapi.services.user.api.model.CreateRequest;
+import br.com.deveficiente.bolaoapi.services.user.api.model.ErrorResponse;
+import br.com.deveficiente.bolaoapi.services.user.api.model.UserResponse;
+import br.com.deveficiente.bolaoapi.services.user.core.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<UserResponse>> index() {
+        List<UserResponse> users = this.userService.getAll().stream()
+                .map(user -> UserResponse.fromUser(user))
+                .collect(Collectors.toList());
+
+        return ResponseEntity.ok(users);
+    }
+
+    @PostMapping
+    public ResponseEntity create(@RequestBody @Valid CreateRequest request) {
```

Comment:
*Protegemos as bordas do sistema como se não houvesse amanhã.* + *Quanto mais externa a borda, mais proteção temos.*

Aqui você já pediu para o Spring MVC proteger a borda aplicando a validação. 

*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*
Como você não pediu o BindingResult, ele vai já vai bloquear a request e retorna um 400 com o erro no formato JSON. Para ser sincero eu chutaria que ele nem cairia nos seus catches.  E cada catch que você adiciona aumenta um ponto na carga cognitiva do seu código(dificuldade de entendimento), então eu seria bem criterioso para colocar um catch que já é feito pelo framework. 

Minha sugestão é tirar o try/catch e deixar só o código que salva.

Reply:
Exatamente, não entra nos catches.
No máximo, vai "surgir" algum tipo de verificação assim quando houver a verificação de login em duplicidade (mas "Aqui não tem espaço para feeling.").

Obrigado!